pkg/controllers/provisioning/v1alpha1/reallocator/controller.go

```diff
@@ -53,31 +55,60 @@ func (c *Controller) Name() string {
 func NewController(kubeClient client.Client, cloudProvider cloudprovider.Factory) *Controller {
 	return &Controller{
 		filter:        &Filter{kubeClient: kubeClient},
+		collector:     &Collector{kubeClient: kubeClient},
 		cloudProvider: cloudProvider,
 	}
 }
 
 // Reconcile executes an allocation control loop for the resource
 func (c *Controller) Reconcile(object controllers.Object) error {
+	provisioner := object.(*v1alpha1.Provisioner)
 	ctx := context.TODO()
-	// 1. Filter all nodes with Karpenter TTL label
-	// TODO: Filter only nodes with the Provisioner label
 
-	// 2. Remove TTL from nodes with TTL label that have pods and if past TTL, cordon/drain node
+	// 1. Filter all nodes with Karpenter labels
+	ttlNodes, err := c.filter.getNodesWithLabels(ctx, []string{
+		provisioning.ProvisionerNamespaceLabelKey,
+		provisioning.ProvisionerNameLabelKey,
+	})
+	if err != nil {
+		return fmt.Errorf("listing ttl provisioner nodes, %w", err)
+	}
+
+	// 2. If node has TTL annotation, and time is after TTL, delete node
+	// TODO: instead of delete - drain then delete
+	for _, node := range ttlNodes.Items {
+		if err := c.collector.ParseTTL(ctx, &node); err != nil {
```

Comment:
Quite unclear what's going on here from the outside. What's being parsed? What's the result of it?